widgets/data_editor.py

```diff
@@ -34,6 +34,16 @@ def load_parameter_names(objectname):
         print(err)
         return None, None
 
+def load_default_values(objectid):
```

Comment:
The function looks quite similar to `route_info()`.

We could consider putting it in a common place (e.g `libbol.py` as a class function of the `MapObject` class), with an extra parameter for choosing between `"DefaultValues"` and `"RouteInfo"`.

And we could implement memoization so that the JSON files are not read/parsed repeatedly. (It can be added at a later time, though.)

Reply:
There are five things in the .json files now, and I don't like the idea of returning 5 things from a function. I would like to just return the contents of the .json file as-is (in case more stuff is added), but I don't know how controversial that decision would be.

Reply:
I think it'd be reasonable to return the `json_data` object (function would be renamed). Then the caller can choose to retrieve `"DefaultValues"` or `"RouteInfo"`.

For example:

```python
def get_object_parameters_json(objectid):
    if not objectid in OBJECTNAMES:
        return None
    name = OBJECTNAMES[objectid]
    filename = os.path.join("object_parameters", name + ".json")
    if not os.path.exists(filename):
        return None
    with open(filename, "r", encoding="utf-8") as f:
        return = json.load(f)
```

```python
json_data = get_object_parameters_json(obj.objectid)
if json_data:
    ... = json_data["DefaultValues"]
```

Reply:
This function could also be used for stuff like tooltips and parameters names. Should those functions be edited to use this new function as well?